src/cuda/search_cuda.h

```diff
@@ -46,6 +46,8 @@ struct GreedySearch_Cuda : Search_Cuda {
   void SampleTopK(int k, float t) override { SampleTopKTopP(k, 0.0, t); }
   void SampleTopP(float p, float t) override { SampleTopKTopP(-1, p, t); }
   void SampleTopKTopP(int k, float p, float t) override;
+  void SetUserTokens(DeviceSpan<int32_t>& next_tokens) override;  // shape (batch_size, sequence_length)
```

Comment:
Should it be AppendUserTokens vs SetUserTokens? Also what does the 'User' part mean?